www/js/diary/cards/TripCard.tsx

```diff
@@ -0,0 +1,173 @@
+/* TripCard displays a card with information about a trip, including a map of the trip route,
+    plus buttons for labeling trips and/or surveying the user about the trip.
+  If the trip has not been processed on the server yet, this is a draft trip, and it
+    will used the greenish/greyish 'draft' theme flavor.
+*/
+
+import React, { useEffect, useState } from "react";
+import { angularize, getAngularService } from "../../angular-react-helper";
+import { View, useWindowDimensions, StyleSheet } from 'react-native';
+import { Card, Divider, IconButton, PaperProvider, Text, useTheme } from 'react-native-paper';
+import { object } from "prop-types";
+import LeafletView from "../../components/LeafletView";
+import { useTranslation } from "react-i18next";
+import TimestampBadge from "./TimestampBadge";
+import MultilabelButtonGroup from "../../survey/multilabel/MultiLabelButtonGroup";
+import UserInputButton from "../../survey/enketo/UserInputButton";
+import useAppConfig from "../../useAppConfig";
+import AddNoteButton from "../../survey/enketo/AddNoteButton";
+import AddedNotesList from "../../survey/enketo/AddedNotesList";
+import { getTheme } from "../../appTheme";
+import { DiaryCard, cardStyles } from "./DiaryCard";
+import { useNavigation } from "@react-navigation/native";
+
+const TripCard = ({ trip }) => {
+
+  const { t } = useTranslation();
+  const { height, width } = useWindowDimensions();
+  const { appConfig, loading } = useAppConfig();
+  const navigation = useNavigation<any>();
+
+  const SurveyOptions = getAngularService('SurveyOptions');
+  const $state = getAngularService('$state');
+
+  const [surveyOpt, setSurveyOpt] = useState(null);
+  const [rerender, setRerender] = useState(false);
+
+  const isDraft = trip.key.includes('UNPROCESSED');
+  const flavoredTheme = getTheme(isDraft ? 'draft' : undefined);
+
+  useEffect(() => {
+    trip.onChanged = () => {
+      console.log("DiaryCard: timelineEntry changed, force update");
+      setRerender(!rerender);
+    }
+  }, []);
+
+  useEffect(() => {
+    const surveyOptKey = appConfig?.survey_info?.['trip-labels'];
+    setSurveyOpt(SurveyOptions[surveyOptKey]);
+  }, [appConfig, loading]);
+
```

Comment:
will we still need `useEffect` when the migration is complete? I assume that the answer is "yes", since we are syncing the config from outside the react ecosystem (over http). Just wanted to check to deepen my understanding.

Reply:
I am not 100% sure, but I think once DynamicConfig is converted we won't have to listen for config changes like this.
It depends how we do it, I think.